src/components/Navigation/TopBar/Notifications/FilterOptions.tsx

```diff
@@ -0,0 +1,283 @@
+  border-radius: ${shape.rounded.borderRadius};
+  border: ${({ $active }) =>
+    $active
+      ? `solid 1px ${colors.interactive.primary__selected_hover.hex}`
+      : `solid 1px ${colors.ui.background__medium.hex}`};
+  background-color: ${({ $active }) =>
+    $active
+      ? colors.interactive.primary__selected_highlight.hex
+      : colors.ui.background__default.hex};
+  color: black;
+  padding: ${spacings.comfortable.x_small} ${spacings.comfortable.small};
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  grid-gap: ${spacings.comfortable.small};
+  cursor: pointer;
+  &:hover {
+    background: ${colors.interactive.primary__selected_highlight.hex};
+  }
+
+  > p {
+    line-height: normal;
+    height: min-content;
+    font-size: 12px;
+  }
+  > svg {
+    color: ${({ $active }) =>
+      $active
+        ? colors.text.static_icons__default.hex
+        : colors.interactive.primary__resting.hex};
+  }
+`;
+
+interface FilterOptionsProps {
+  onFilter: (value: FilterNotification[]) => void;
+  onSort: (value: SortNotification[]) => void;
+  sortMenuRef: MutableRefObject<HTMLDivElement | null>;
+  filterMenuRef: MutableRefObject<HTMLDivElement | null>;
+}
+
+const FilterOptions: FC<FilterOptionsProps> = ({
+  onFilter,
+  onSort,
+  sortMenuRef,
+  filterMenuRef,
+}) => {
+  const [openFilter, setOpenFilter] = useState(false);
+  const [openSort, setOpenSort] = useState(false);
+  const filterRef = useRef<HTMLButtonElement | null>(null);
+  const sortRef = useRef<HTMLButtonElement | null>(null);
+
+  const [selectedSort, setSelectedSort] = useState({
+    [SortNotification.UNREAD]: false,
+    [SortNotification.OLD_NEWEST]: false,
+    [SortNotification.NEW_OLDEST]: false,
+  });
+
+  const [selectedFilter, setSelectedFilter] = useState({
+    [FilterNotification.SYSTEM]: false,
+    [FilterNotification.USER]: false,
+    [FilterNotification.UNREAD]: false,
+  });
+
+  const onClickOpenFilter = () => {
+    setOpenFilter(true);
+    setOpenSort(false);
+  };
+  const onClickCloseFilter = () => {
+    setOpenFilter(false);
+  };
+
+  const onClickOpenSort = () => {
+    setOpenSort(true);
+    setOpenFilter(false);
+  };
+
+  const onClickCloseSort = () => {
+    setOpenSort(false);
+  };
+
+  const handleFilter = (value: FilterNotification) => {
+    onFilter(selectedFilter[value] ? [] : [value]);
+    setSelectedFilter((prevState) => ({
+      ...prevState,
+      [value]: !prevState[value],
+    }));
+  };
+
+  const handleSorting = (value: SortNotification) => {
+    onSort(selectedSort[value] ? [] : [value]);
+    setSelectedSort((prevState) => ({
+      ...prevState,
+      [value]: !prevState[value],
+    }));
+  };
+
+  useOutsideClick(filterMenuRef.current, (event) => {
+    if (
+      openFilter &&
+      filterRef.current !== null &&
+      !filterRef.current?.contains(event.target as Node)
+    ) {
+      onClickCloseFilter();
+    }
+  });
+
+  useOutsideClick(sortMenuRef.current, (event) => {
```

Comment:
Instead of using `useOutsideClick` on these menus you can use the 'onClose' property on the EDS menus, if you don't want the menu to close when clicking a menu item simply use the 'closeMenuOnClick=false' property for the Menu.Item elements 😄

Reply:
Ah thanks :D